### ISSUE TITLE
Feature request: reduce_random

### ISSUE THREAD

**Issue by billzyx** [tensorflow/tensorflow#22983]
Just as `tf.reduce_max`, `tf.reduce_random` is to randomly return the elements across dimensions of a tensor with some probabilities, such as respect to the value of the tensor at those dimensions.

Or any other ways to implement it?

**Comment by ppwwyyxx** [comment]
Could you first define __mathematically__ what is the operation you're requesting? I think it's not clear enough from your text description.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
